tests/integration_rust/pypi_tests.rs

```diff
@@ -82,3 +82,45 @@ async fn test_index_strategy() {
         Some("3.0.0".into())
     );
 }
+
+#[tokio::test]
+#[cfg_attr(not(feature = "slow_integration_tests"), ignore)]
+async fn test_pinning_index() {
+    let pypi_indexes = Path::new(env!("CARGO_MANIFEST_DIR")).join("tests/data/pypi-indexes");
+    let pypi_indexes_url = Url::from_directory_path(pypi_indexes.clone()).unwrap();
+
+    let pixi = PixiControl::from_manifest(&format!(
+        r#"
+        [project]
+        name = "pypi-pinning-index"
+        platforms = ["{platform}"]
+        channels = ["conda-forge"]
+
+        [dependencies]
+        python = "~=3.12.0"
+
+        [pypi-dependencies]
+        foo = {{ version = "*", index = "{pypi_indexes}multiple-indexes-a/index" }}
```

Comment:
I think it's good to also add torch here, it shouldn't be to slow as it's only solving and it's the main usecase for this feature.

Reply:
I've added this in different test, I think its better to isolate this one :)